doc: Documentation review for Azure documentation (#267)
bifravst/bifravst: docs/azure/DeviceCredentials.rst

.. _azure-device-credentials:

Device credentials
##################

On Azure devices, connect to IoT Hub using `TLS version 1.2 <https://docs.microsoft.com/en-us/azure/iot-fundamentals/iot-security-deployment>`_ and Elliptic Curve Cryptography (ECC) based certificates.

Azure supports offline generation of device certificates through `Enrollment Groups <https://docs.microsoft.com/bs-latn-ba/azure/iot-dps/quick-enroll-device-x509-node>`_.

Device certificates are signed using an intermediate CA.
This allows to keep the CA root certificate secure and create an intermediate CA certificate to be used during device provisioning.

> In this case, CA root certificate private key will never be transmitted to insecure locations (for example, a third-party factory).

Reviewer: ```suggestion
Because of that, CA root certificate private key will never be transmitted to insecure locations (for example, a third-party factory).
```

This is not referring to different cases/scenarios, it's a reference to the decision of "using an intermediate CA."

If the intermediate CA certificate is compromised, it can be deactivated.
New devices with credentials signed by the compromised intermediate CA certificate will no longer be provisioned.

Devices that previously connected to the IoT Hub, will continue to work.

> For creating device credentials, you must generate the following certificates:

Reviewer: I find this a little confusing, because the next section is about the DPS.

![image](https://user-images.githubusercontent.com/188915/106742262-1954ba80-661d-11eb-9a38-42d25ce63db7.png)

So how about you move that list of steps before the individual sections?

## Creating device credentials

For creating device credentials, you must generate the following certificates:

- CA root certificate
- CA intermediate certificate
- Device certificate

### CA root certificate

... 

### CA intermediate certificate

... 

### Device certificate

...

* CA root certificate
* CA intermediate certificate
* Device certificate

Device Provisioning Service
***************************

Devices connect to IoT Hub using the `Device Provisioning Service (DPS) <https://docs.microsoft.com/en-us/azure/iot-dps/>`_.
DPS allows better management of roaming devices.
However, it requires an extra connection step.
Following are the overall connection steps:

* Connect to DPS to retrieve the endpoint configuration
* Connect to the IoT Hub endpoint.

This is a one-time operation, and the device can cache its associated IoT Hub endpoint.

Create a CA root certificate
****************************

.. note::

>     Make sure that you have exported the right resource group name as ``$RESOURCE_GROUP_NAME``.
> 	By default, it is set to ``bifravst``.

Reviewer: ```suggestion
    Make sure that you have exported the right resource group name as ``$RESOURCE_GROUP_NAME``.
    By default, it is set to ``bifravst``.
```

Removes indentation.

![Uploading image.png…]()
